components/google-cloud/google_cloud_pipeline_components/experimental/HyperparameterTuningJob/component.py

```diff
@@ -0,0 +1,213 @@
+import json
+
+from kfp.v2.google.client import AIPlatformClient
+from kfp.components import create_component_from_func_v2
+from google.cloud import aiplatform
+from google_cloud_pipeline_components.aiplatform import utils
+from typing import NamedTuple, Optional, Union, Sequence, Dict, List, Any
+
+
+def hyperparameter_tuning_job(
+    display_name: str,
+    project: str,
+    location: str,
+    staging_bucket: str,
+    worker_pool_specs: List[Dict],
+    metric_spec: Dict[str, str],
+    parameter_spec: Dict[str, Any],
+    max_trial_count: int,
+    parallel_trial_count: int,
+    max_failed_trial_count: int = 0,
+    search_algorithm: Optional[str] = None,
+    measurement_selection: Optional[str] = "best",
+    encryption_spec_key_name: Optional[str] = None,
+    base_output_dir: Optional[str] = None,
+    local_script_kwargs: Optional[Dict] = None,
+) -> NamedTuple('Outputs', [
+    ("trials", list),
+]):
+    """
+    Creates a Google Cloud AI Plaform HyperparameterTuning Job.
+
+    For more information on using hyperparameter tuning please visit:
+    https://cloud.google.com/ai-platform-unified/docs/training/using-hyperparameter-tuning
+
+    Args:
+        display_name (str):
+            Required. The user-defined name of the HyperparameterTuningJob.
+            The name can be up to 128 characters long and can be consist
+            of any UTF-8 characters.
+        project (str):
+            Required. Project to run the HyperparameterTuningJob in.
+        location (str):
+            Required. Location to run the HyperparameterTuningJob in.
+        staging_bucket (str):
+            Required. Bucket for produced HyperparameterTuningJob artifacts.
+        worker_pool_specs (List[Dict]):
+            Required. The spec of the worker pools including machine type and
+            Docker image, as a list of dictionaries.
+        metric_spec: Dict[str, str]
+            Required. Dicionary representing metrics to optimize. The dictionary key is the metric_id,
+            which is reported by your training job, and the dictionary value is the
+            optimization goal of the metric('minimize' or 'maximize'). example:
+            metric_spec = {'loss': 'minimize', 'accuracy': 'maximize'}
+        parameter_spec (Dict[str, hyperparameter_tuning._ParameterSpec]):
```

Comment:
It seems to me that this would fail as the KFP SDK wouldn't be able to serialize and deserialize: [_ParameterSpec](https://github.com/googleapis/python-aiplatform/blob/master/google/cloud/aiplatform/hyperparameter_tuning.py#L33).

I think we need an intermediate json representation for _ParameterSpec.